Core/include/Acts/Propagator/GuidedNavigator.hpp

```diff
@@ -0,0 +1,230 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Geometry/GeometryIdentifier.hpp"
+#include "Acts/Geometry/TrackingGeometry.hpp"
+#include "Acts/Geometry/TrackingVolume.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+namespace Acts {
+
+template <typename Guider>
+class GuidedNavigator {
+  Guider m_guider;
+  std::shared_ptr<const TrackingGeometry> m_tgeo;
+
+ public:
+  struct State {
+    std::vector<SurfaceIntersection> navSurfaces;
+    std::vector<SurfaceIntersection>::const_iterator navSurfaceIter;
+    const Surface *currentSurface;
+    const Surface *startSurface;
+
+    bool navigationBreak = false;
+    bool targetReached = false;
+
+    const TrackingVolume *currentVolume = nullptr;
+    const TrackingVolume *worldVolume = nullptr;
+    const Surface *targetSurface = nullptr;
+  };
+
+  // TODO Must the Navigator really be default constructable?
+  // See Propagator.hpp:247
+  GuidedNavigator() { throw_assert(false, "Don't use this constructor!"); }
```

Comment:
That might be something we should get rid of, actually.

Reply:
You mean the requirement that a navigator needs to be default constructable? Because at least I would say that a default constructed `Acts::Navigator` isn't really useful either...